extras/userdoc/generator/modules/writers.py

```diff
@@ -0,0 +1,289 @@
+
+            htmllist.append('<b>Name:</b><pre>%s - %s</pre>' %
+                            (name, fullname))
+            hlpfullname = re.sub(' <br\/> ', '\n', fullname).strip()
+            hlplist.append('Name: %s - %s\n' % (name, hlpfullname))
+
+    # print keywords
+    # sorting linked keywords
+    for word in keywords:
+        word = word.strip(':')
+        for key, value in doc_dic.iteritems():
+            if key == word:
+                if (key != "Name" and key != "FullName" and key != "SeeAlso"
+                   and key != "File"):
+                    value = re.sub("(######)", " <br/> ", value)
+                    # value = re.sub("(\~\~)", '  ', value)
+                    value = re.sub("(\~\~\~)", '\t', value)
+                    value = re.sub("(\~\~)", '  ', value)
+                    value = re.sub("(\~)", ' ', value)
+                    value = re.sub(' - ', '\t- ', value)
+
+                    htmllist.append('<b>%s: </b>' % key)
+                    htmllist.append('<pre>%s</pre>' % value)
+                    hlpvalue = re.sub(' <br\/> ', '\n', value).rstrip()
+                    hlpvalue = re.sub('\n ', '\n', hlpvalue).rstrip()
+                    hlpvalue = hlpvalue.lstrip('\n')
+                    hlpvalue = re.sub('\n[\s?]*\n', '\n', hlpvalue).rstrip()
+                    # Better looking .hlp files
+                    dedented_text = textwrap.dedent(hlpvalue).strip()
+                    hlpcontent = ('%s:\n\n%s\n\n' % (key, dedented_text))
+                    hlplist.append(hlpcontent)
+
+    for key, value in doc_dic.iteritems():
+        if key == "SeeAlso":
+            htmllist.append('<b>%s: </b>' % key)
+            hlplist.append('%s:\n' % key)
+            htmllist.append('<ul>')
+            for i in value:
+                see = i.strip("###### ###### $$")
+                see = i.strip("###### ~~")
+                if see:
+                    if see in sli_command_list:
+                        htmllist.append('    <li><a href="../sli/' + see +
+                                        '.html">' + see + '</a></li>')
+                        hlplist.append('%s' % see)
+                    else:
+                        htmllist.append('    <li><a href="../cc/' + see +
+                                        '.html">' + see + '</a></li>')
+                        hlplist.append('%s' % see)
+            hlplist.append('')
+            htmllist.append('</ul>')
+
+    for key, value in doc_dic.iteritems():
+        if key == "File":
+            value = value.strip("###### ###### $$")
+            htmllist.append('<b>Source:</b><pre>%s</pre>' % value)
+            hlplist.append('Source:\n\n%s' % value)
+    htmllist.append('</div>')
+    htmlstring = ('\n'.join(htmllist))
+    cmdindexstring = s.substitute(indexbody=htmlstring, css=csstempl,
+                                  title=name, footer=footertempl)
+
+    if name:  # only, if there is a name
```

Comment:
Aren't these two following one piece of code, but just with a difference in the sli versus cc element of the filename?